client/web/src/repo/settings/RepoSettingsOptionsPage.tsx

```diff
@@ -50,6 +66,22 @@ export const RepoSettingsOptionsPage: FC<Props> = ({ repo, history }) => {
         SiteExternalServiceConfigVariables
     >(SITE_EXTERNAL_SERVICE_CONFIG, {})
 
+    const [excludeRepo, { data: excludeData, error: excludeError, loading: isExcluding }] = useMutation<
+        ExcludeRepoFromExternalServicesResult,
+        ExcludeRepoFromExternalServicesVariables
+    >(EXCLUDE_REPO_FROM_EXTERNAL_SERVICES, {
+        onCompleted: () => {
+            let count = 3
+            setInterval(() => {
+                if (count === 0) {
+                    history.push('/site-admin/external-services')
+                }
+                setTtl(count)
+                count--
+            }, 700)
```

Comment:
Does it make sense to clear the interval after we do not need it anymore?

```suggestion
            let interval = setInterval(() => {
                if (count === 0) {
                    clearInterval(interval)
                    history.push('/site-admin/external-services')
                }
                setTtl(count)
                count--
            }, 700)
```

Reply:
We should definitely clean it up! +1 on this

Reply:
My first usage of this thing, TIL about clearing it! Thanks, guys!